lib/timeline/component/item/BoxItem.js

```diff
@@ -58,63 +58,57 @@ BoxItem.prototype.isVisible = function(range) {
   return isVisible;
 };
 
-/**
- * Repaint the item
- */
-BoxItem.prototype.redraw = function() {
-  var dom = this.dom;
-  if (!dom) {
+BoxItem.prototype._createDomElement = function() {
+  if (!this.dom) {
     // create DOM
     this.dom = {};
-    dom = this.dom;
 
     // create main box
-    dom.box = document.createElement('DIV');
+    this.dom.box = document.createElement('DIV');
 
     // contents box (inside the background box). used for making margins
-    dom.content = document.createElement('DIV');
-    dom.content.className = 'vis-item-content';
-    dom.box.appendChild(dom.content);
+    this.dom.content = document.createElement('DIV');
+    this.dom.content.className = 'vis-item-content';
+    this.dom.box.appendChild(this.dom.content);
 
     // line to axis
-    dom.line = document.createElement('DIV');
-    dom.line.className = 'vis-line';
+    this.dom.line = document.createElement('DIV');
+    this.dom.line.className = 'vis-line';
 
     // dot on axis
-    dom.dot = document.createElement('DIV');
-    dom.dot.className = 'vis-dot';
+    this.dom.dot = document.createElement('DIV');
+    this.dom.dot.className = 'vis-dot';
 
     // attach this item as attribute
-    dom.box['timeline-item'] = this;
+    this.dom.box['timeline-item'] = this;
 
     this.dirty = true;
   }
+}
 
-  // append DOM to parent DOM
+BoxItem.prototype._appendDomElement = function() {
   if (!this.parent) {
     throw new Error('Cannot redraw item: no parent attached');
   }
-  if (!dom.box.parentNode) {
+  if (!this.dom.box.parentNode) {
     var foreground = this.parent.dom.foreground;
     if (!foreground) throw new Error('Cannot redraw item: parent has no foreground container element');
-    foreground.appendChild(dom.box);
+    foreground.appendChild(this.dom.box);
   }
-  if (!dom.line.parentNode) {
+  if (!this.dom.line.parentNode) {
     var background = this.parent.dom.background;
     if (!background) throw new Error('Cannot redraw item: parent has no background container element');
-    background.appendChild(dom.line);
+    background.appendChild(this.dom.line);
   }
-  if (!dom.dot.parentNode) {
+  if (!this.dom.dot.parentNode) {
     var axis = this.parent.dom.axis;
     if (!background) throw new Error('Cannot redraw item: parent has no axis container element');
-    axis.appendChild(dom.dot);
+    axis.appendChild(this.dom.dot);
   }
   this.displayed = true;
+}
 
-  // Update DOM when item is marked dirty. An item is marked dirty when:
-  // - the item is not yet rendered
-  // - the item's data is changed
-  // - the item is selected/deselected
+BoxItem.prototype._updateDirtyDomComponents = function() {
```

Comment:
As with `BackGroundItem.updateDirtyDomComponents()`, please retain commenting. It is useful.